test/object-store/sync/sync_test_utils.cpp

```diff
@@ -236,22 +236,22 @@ void wait_for_advance(Realm& realm)
     realm.m_binding_context = nullptr;
 }
 
-std::pair<ThreadSafeReference, std::exception_ptr> async_open_realm(const Realm::Config& config)
+void async_open_realm(const Realm::Config& config,
+                      util::UniqueFunction<void(ThreadSafeReference&& ref, std::exception_ptr e)> finish)
 {
     std::mutex mutex;
-    ThreadSafeReference realm_ref;
-    std::exception_ptr error;
+    bool did_finish = false;
     auto task = Realm::get_synchronized_realm(config);
-    task->start([&](ThreadSafeReference&& ref, std::exception_ptr e) {
+    task->start([&, callback = std::move(finish)](ThreadSafeReference&& ref, std::exception_ptr e) {
+        callback(std::move(ref), e);
```

Comment:
What if `callback` is a nullptr?
Maybe `REALM_ASSERT(callback)` or at least check for nullptr?